src/bidiMapper/utils/Tests/stub.spec.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright 2021 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { assert, createStubInstance, SinonStubbedInstance } from 'sinon';
+
+export class Stub {
+  // Returns mock object of a given type.
+  static stub<T>(constructor: new (...args) => T): StubbedClass<T> {
```

Comment:
What are the advantages to stub/StubbedClass<T> over using createStubInstance directly?

Reply:
The advantage is in casting the result to `StubbedClass`, which is simultaneously `SinonStubbedInstance<T>`  and `T` itself, which allows to avoid disturbing casting in each stub usage.

Reply:
The idea was to make a stubbed instance of type `SinonStubbedInstance<T> | T`, which would allow to both use it as a parameter for downstream classes and to verify stubs. For now `StubServer` solves the issue by making it's methods both stubs and methods.